src/local/room_xy/extra_math.rs

```diff
@@ -0,0 +1,167 @@
+    /// let pos2 = pos1 + (7, 7);
+    /// assert_eq!(pos2, RoomXY::checked_new(49, 49).unwrap());
+    /// ```
+    #[inline]
+    #[track_caller]
+    fn add(self, (x, y): (i8, i8)) -> Self {
+        self.checked_add((x, y)).unwrap()
+    }
+}
+
+impl Add<Direction> for RoomXY {
+    type Output = RoomXY;
+
+    /// Adds a `Direction` to this position's coordinates.
+    ///
+    /// # Panics
+    ///
+    /// Will panic if the new position is outside standard room bounds.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// # use screeps::{RoomXY, Direction};
+    ///
+    /// let pos1 = RoomXY::checked_new(49, 40).unwrap();
+    /// let pos2 = pos1 + Direction::Top;
+    /// assert_eq!(pos2, RoomXY::checked_new(49, 39).unwrap());
+    /// ```
+    #[inline]
+    #[track_caller]
+    fn add(self, direction: Direction) -> Self {
+        self.checked_add_direction(direction).unwrap()
+    }
+}
+
+impl Sub<(i8, i8)> for RoomXY {
+    type Output = RoomXY;
+
+    /// Subtracts an `(x, y)` pair from this position's coordinates.
+    ///
+    /// # Panics
+    ///
+    /// Will panic if the new position is outside standard room bounds.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// # use screeps::RoomXY;
+    ///
+    /// let pos1 = RoomXY::checked_new(49, 40).unwrap();
+    /// let pos2 = pos1 - (49, 0);
+    /// assert_eq!(pos2, RoomXY::checked_new(0, 40).unwrap());
+    /// ```
+    #[inline]
+    #[track_caller]
+    fn sub(self, (x, y): (i8, i8)) -> Self {
+        self.checked_add((-x, -y)).unwrap()
+    }
+}
+
+impl Sub<Direction> for RoomXY {
+    type Output = RoomXY;
+
+    /// Subtracts a `Direction` from this position's coordinates.
+    ///
+    /// # Panics
+    ///
+    /// Will panic if the new position is outside standard room bounds.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// # use screeps::{RoomXY, Direction};
+    ///
+    /// let pos1 = RoomXY::checked_new(49, 40).unwrap();
+    /// let pos2 = pos1 - Direction::Top;
+    /// assert_eq!(pos2, RoomXY::checked_new(49, 41).unwrap());
+    /// ```
+    #[inline]
+    fn sub(self, direction: Direction) -> Self {
+        self.checked_add_direction(-direction).unwrap()
+    }
+}
+
+impl Sub<RoomXY> for RoomXY {
+    type Output = (i8, i8);
+
+    /// Subtracts the other position from this one, extracting the
+    /// difference as the output.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// # use screeps::RoomXY;
+    ///
+    /// let pos1 = RoomXY::checked_new(40, 40).unwrap();
+    /// let pos2 = RoomXY::checked_new(0, 20).unwrap();
+    /// assert_eq!(pos1 - pos2, (40, 20));
+    /// ```
+    #[inline]
+    fn sub(self, other: RoomXY) -> (i8, i8) {
+        let mx: i8 = self.x.0.try_into().unwrap();
+        let my: i8 = self.y.0.try_into().unwrap();
+        let ox: i8 = other.x.0.try_into().unwrap();
+        let oy: i8 = other.y.0.try_into().unwrap();
+
+        ((mx - ox), (my - oy))
```

Comment:
Why not something like `(self.x.0.wrapping_sub(other.x.0) as i8, self.y.0.wrapping_sub(other.y.0) as i8)`?

Reply:
Because it's not clear at first glance (or even second glance) that using `wrapping_sub` and then casting to `i8` is going to give you the proper negative offset. That didn't even come to mind, compared to just ensuring the arguments are all `i8` from the get-go and then subtracting them.

Reply:
You're adding 4 extra comps and panic infrastructure to implement sub like this, though. And wrapping sub working like this is the entire reason signed arithmetic works between positive signed integers.

Reply:
That's fair. For a core library like this, efficiency does matter more than code clarity.